### PR TITLE
[cozy-doctypes] Contacts: improve getInitials

### PR DESCRIPTION
**Pull request by cedricmessiant**
- fix getInitals when contact has only givenName or familyName
- use the primary email's first letter when contact has no name

### CODE REVIEW

File: packages/cozy-doctypes/src/contacts/Contact.js [hunk 1]
```diff
@@ -22,11 +22,22 @@ class Contact extends Document {
...
+        .toUpperCase()
+    }
+
+    const email = Contact.getPrimaryEmail(contact)
+    if (email) {
+      return email[0].toUpperCase()
+    }
+
+    // eslint-disable-next-line no-console
+    console.error('Contact has no name and no email.')
```
Comment (Crash--) [inline packages/cozy-doctypes/src/contacts/Contact.js]: Utiliser log() ? https://github.com/cozy/cozy-libs/blob/master/packages/cozy-doctypes/src/log.js

File: packages/cozy-doctypes/src/contacts/Contact.spec.js [hunk 2]
```diff
@@ -13,6 +13,40 @@ describe('Contact model', () => {
...
+        email: [
+          {
+            address: 'arya.stark@thenorth.westeros',
+            primary: true
+          }
+        ]
+      }
+      const result = Contact.getInitials(contact)
+      expect(result).toEqual('A')
+    })
```
Comment (Crash--) [inline packages/cozy-doctypes/src/contacts/Contact.spec.js]: it should return '' if no name, no email ? ;)

**Comment by cedricmessiant** [comment]
I think we can remove the support for string param to `getInitials`, WDYT?

File: packages/cozy-doctypes/src/contacts/Contact.js [hunk 3]
```diff
@@ -15,18 +16,28 @@ class Contact extends Document {
...
     }
 
-    return ['givenName', 'familyName']
-      .map(part => contact.name[part] || '')
-      .map(name => name[0])
-      .join('')
-      .toUpperCase()
+    if (contact.name) {
+      return ['givenName', 'familyName']
+        .map(part => (contact.name && contact.name[part]) || '')
```
Comment (y-lohse) [inline packages/cozy-doctypes/src/contacts/Contact.js]: Nit: lodash.get would probably be easier here :)

**Pull request merged**